### PR TITLE
Add presn.Yoshida_2016 model

### LINKED ISSUE #215: Add presn.Yoshida model
Implement Yoshida model from https://zenodo.org/record/3778014

### PR DESCRIPTION
**Pull request by Sheshuk**
Closes #215 

- [x] Add loader class
- [x] Add registry model 
- [x] Add example notebook
- [x] Cross-check the resulting  plots with the model paper

### CODE REVIEW

**Comment by Sheshuk** [comment]
Cross-check with Yoshida_2016 [paper](https://arxiv.org/abs/1606.04915) (Fig.5):
image

Using a script here I plot the image 
https://gist.github.com/Sheshuk/3a11f9b033fb8a3029b53fada9aa4f9a
image

**Comment by JostMigenda** [comment]
From the figure, it looks like the NU_E and NU_X rates are almost identical (both solid lines overlap); while the NU_E_BAR and NU_X_BAR are both almost identical to each other (both dashed lines overlap), and a factor of ~4 lower than the solid lines. However, physically, this doesn’t seem right at all: The main emission mechanism is pair production; so NU_E should be very similar to NU_E_BAR, NU_X should be very similar to NU_X_BAR, and there’s no a priori reason why NU_E and NU_X should be similar.

Therefore, I suspect that NU_E_BAR and NU_X have been mixed up here. Looking at the README, the input files use the order NU_E, NU_E_BAR, NU_X, NU_X_BAR; while the `Flavor` enum in `snewpy.neutrino` uses the order NU_E, NU_X, NU_E_BAR, NU_X_BAR. Maybe that’s the origin of the problem?

File: python/snewpy/models/presn_loaders.py [hunk 1]
```diff
@@ -139,3 +139,36 @@ def get_initial_spectra(self, t, E, flavors=Flavor):
...
+    """
+    def __init__(self, filename):
+        with open(self.request_file(filename)) as f:
+            data = []
+            T = []
+            while (line := f.readline()):
+                if(not line):break
+                N_step,T_step,logT_c,logRho_c = [conv(token) for conv,token in zip([int,float,float,float],line.split())]
+                data += [[np.loadtxt(f, max_rows=100).flatten() for i in range(4)]]
+                T+=[T_step]
```
Comment (JostMigenda) [inline python/snewpy/models/presn_loaders.py:154]:
```suggestion
                if not line: break
                T += [float(line.split()[1])]
                data += [[np.loadtxt(f, max_rows=100).flatten() for i in range(4)]]
```

**Review by JostMigenda — CHANGES_REQUESTED** [review]
Looks good to me now; but let’s add this to the `test_presn_rates.py` to have at least some basic checks that the model initializes successfully and produces reasonable rates.

File: python/snewpy/models/presn_loaders.py (hunk 1 above)
Comment (Sheshuk) [reply python/snewpy/models/presn_loaders.py:154]: This adds a "magic number" `[1]` here, so for me it's less clear to see, what is going on.
But it's subjective. So if it makes code look better to you, let's do it

**Review by JostMigenda — APPROVED** [review]
Looks good; thanks a lot! ☺️

**Pull request merged**